result.test.ts

```diff
@@ -0,0 +1,146 @@
+        const error = new Error("test error");
+        const result = failure(error);
+        assertEquals(result.error, error);
+        assertEquals(result.error.message, "test error");
+      });
+    });
+
+    describe("edge case testing", () => {
+      const edgeCases = [
+        {
+          name: "object value",
+          error: { message: "custom error occurred" },
+          description: "will return the same object as error",
+        },
+        {
+          name: "null value",
+          error: null,
+          description: "will return null as error",
+        },
+        {
+          name: "undefined value",
+          error: undefined,
+          description: "will return undefined as error",
+        },
+        {
+          name: "CustomError instance",
+          error: new CustomError("custom error occurred"),
+          description: "will return the CustomError instance as error",
+        },
+      ];
+
+      edgeCases.forEach(({ name, error, description }) => {
+        it(`should handle ${name} - ${description}`, () => {
+          const result = failure(error as Error);
+          assertEquals(result.error, error);
+        });
+      });
+    });
+  });
+
+  describe("isSuccess", () => {
+    describe("happy path", () => {
+      const happyCases = [
+        {
+          name: "on success result",
+          result: success(100),
+          expected: true,
+          description: "should return true for success result",
+        },
+        {
+          name: "on failure result",
+          result: failure(new Error("failure")),
+          expected: false,
+          description: "should return false for failure result",
+        },
+      ];
+
+      happyCases.forEach(({ name, result, expected, description }) => {
+        it(`should handle ${name} - ${description}`, () => {
+          assertEquals(isSuccess(result), expected);
+          if (isSuccess(result)) {
+            assertEquals("data" in result, true);
+          } else {
+            assertEquals("error" in result, true);
+          }
+        });
+      });
+    });
+  });
+
+  describe("isFailure", () => {
+    describe("happy path", () => {
+      const happyCases = [
+        {
+          name: "on failure result",
+          result: failure(new Error("failure")),
+          expected: true,
+          description: "should return true for failure result",
+        },
+        {
+          name: "on success result",
+          result: success(100),
+          expected: false,
+          description: "should return false for success result",
+        },
+      ];
+
+      happyCases.forEach(({ name, result, expected, description }) => {
+        it(`should handle ${name} - ${description}`, () => {
+          assertEquals(isFailure(result), expected);
+          if (isFailure(result)) {
+            assertEquals("error" in result, true);
+          } else {
+            assertEquals("data" in result, true);
+          }
```

Comment:
[imo]
シンプルに記述できそうです。
```suggestion
          assertEquals("error" in result, isFailure(result));
```